shared/src/business/useCases/unsealCaseInteractor.test.js

```diff
@@ -0,0 +1,35 @@
+const { applicationContext } = require('../test/createTestApplicationContext');
+const { MOCK_CASE } = require('../../test/mockCase');
+const { ROLES } = require('../entities/EntityConstants');
+const { unsealCaseInteractor } = require('./unsealCaseInteractor');
+
+describe('unsealCaseInteractor', () => {
+  beforeAll(() => {
+    applicationContext
+      .getPersistenceGateway()
+      .getCaseByDocketNumber.mockReturnValue(MOCK_CASE);
+  });
+
+  it('should throw an error if the user is unauthorized to unseal a case', async () => {
+    await expect(
+      unsealCaseInteractor(applicationContext, {
+        docketNumber: MOCK_CASE.docketNumber,
+        qcCompleteForTrial: true,
+        trialSessionId: '10aa100f-0330-442b-8423-b01690c76e3f',
+      }),
+    ).rejects.toThrow('Unauthorized for unsealing cases');
+  });
```

Comment:
Kinda seems like this test takes for granted who applicationContext is. Do you think it would make sense to mock `petitionsClerk` (or even all of the roles that aren't `docketClerk` and `clerkOfCourt`) to this test? Is that overkill?

Reply:
Great idea!  I made this test to specifically use the `petitionsClerk`.  I think if we wanted to test all the roles, we would want a specific test of `authorizationClientService.isAuthorized` or at least that a specific role's permision contains a specific `authorizationClientService.ROLE_PERMISSIONS`.  But then, we might just be recreating the array logic used inside the thing we're testing (which is `authorizationClientService`).  🤔